src/frontend/src/featureFlags/index.ts

```diff
@@ -0,0 +1,62 @@
+// Feature flags with default values
+const FEATURE_FLAGS_WITH_DEFAULTS = {
+  DOMAIN_COMPATIBILITY: false,
+} as const satisfies Record<string, boolean>;
+
+const LOCALSTORAGE_FEATURE_FLAGS_PREFIX = "ii-localstorage-feature-flags__";
+
+export class FeatureFlag {
+  readonly #storage: Pick<Storage, "getItem" | "setItem" | "removeItem">;
+  readonly #key: string;
+  readonly #defaultValue: boolean;
+  #value: boolean;
+
+  constructor(
+    storage: Pick<Storage, "getItem" | "setItem" | "removeItem">,
+    key: string,
+    defaultValue: boolean
+  ) {
+    this.#storage = storage;
+    this.#key = key;
+    this.#defaultValue = defaultValue;
+    const storedValue = this.#storage.getItem(this.#key);
+    this.#value =
+      storedValue === null
+        ? this.#defaultValue
+        : Boolean(JSON.parse(storedValue));
```

Comment:
Do we want to add a try/catch here in case there is a problem parsing the stored value? And revert to the default value instead?

Reply:
Good catch, will add it.